org.eclipse.scanning.event/src/org/eclipse/scanning/event/remote/_Scannable.java

```diff
@@ -253,6 +254,7 @@ public MonitorRole getMonitorRole() {
 	public MonitorRole setMonitorRole(MonitorRole role) throws ScanningException {
 		if (info==null) update();
 		MonitorRole oldRole = info.getMonitorRole();

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "info" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)